core/domain/activity_jobs_one_off.py

```diff
@@ -589,3 +589,154 @@ def map(snapshot_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class ValidateSnapshotMetadataModelsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that validates the commit logs of the SnapshotMetadataModels.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        # Considering all the models that has a SnapshotMetadata except
+        # config model because it does not have commit logs.
+        return [
+            collection_models.CollectionSnapshotMetadataModel,
+            collection_models.CollectionRightsSnapshotMetadataModel,
+            exp_models.ExplorationSnapshotMetadataModel,
+            exp_models.ExplorationRightsSnapshotMetadataModel,
+            question_models.QuestionSnapshotMetadataModel,
+            skill_models.SkillSnapshotMetadataModel,
+            story_models.StorySnapshotMetadataModel,
+            topic_models.TopicSnapshotMetadataModel,
+            topic_models.SubtopicPageSnapshotMetadataModel,
+            topic_models.TopicRightsSnapshotMetadataModel
+        ]
+
+    @staticmethod
+    def map(snapshot_model):
+
+        class_name = snapshot_model.__class__.__name__
+        missing_commit_log_msg = 'MISSING COMMIT LOGS-'
+
+        if isinstance(
```

Comment:
Can you remove the if/else by finding the last "-" and taking the substring before that?

If you like, you can also validate that the part after the last "-" is a number, and yield an error message (and exit) if it's not.

Reply:
Done.